mlflow/openai/__init__.py

```diff
@@ -558,12 +574,47 @@ def predict(
             }
             for messages in messages_list
         ]
+        results = process_api_requests(requests)
+        return [r["choices"][0]["message"]["content"] for r in results]
+
+    def _predict_embeddings(self, data):
+        import openai
+
+        kwargs = self.model.copy()
+        kwargs.pop("task", None)
+        first_string_column = _first_string_column(data)
+        texts = data[first_string_column].tolist()
+        res = []
+        batch_size = 20
```

Comment:
shall we make batch_size configurable ?

Reply:
If there is a request.

Reply:
Randomly chose 20. A larger value (say 100?) might be faster.

Reply:
What does OpenAI support per batch? We should use the maximum thing they support, IMO :)

Reply:
As we discussed, updated to 1024.